storm-client/src/jvm/org/apache/storm/windowing/TimeTriggerPolicy.java

```diff
@@ -62,7 +63,9 @@ public void reset() {
 
     @Override
     public void start() {
-        executorFuture = executor.scheduleAtFixedRate(newTriggerTask(), duration, duration, TimeUnit.MILLISECONDS);
+        // initial delay is slightly less than the duration so that the initial tuples wont't expire due to time drift
+        long initialDelay = duration - Math.min((long) (duration * .1), 100);
+        executorFuture = executor.scheduleAtFixedRate(newTriggerTask(), initialDelay, duration, TimeUnit.MILLISECONDS);
```

Comment:
We cant't guarantee the absolute time at which the trigger will fire since it depends on when and how long storm takes to initialize the bolt. Only the relative time difference between two window firings can be guaranteed. The issue was that the first trigger was firing a few milliseconds after the duration which was causing the initial tuples to directly expire. I have refactored the logic to not depend on the initial delay being slightly lesser.